internal_packages/keybase/lib/modal-key-recommender.cjsx

```diff
@@ -68,8 +68,23 @@ class ModalKeyRecommender extends React.Component
         ['Encrypt', 'Cancel']
       )
         return
-    Actions.closePopover()
-    @props.callback(@state.identities)
+
+    if _.every(@state.identities, (identity) -> identity.key?)
+      Actions.closePopover()
+      @props.callback(@state.identities)
+    else
+      emptyIdents = _.filter(@state.identities, (identity) -> !identity.key?)
+      newIdents = []
+      for idIndex of emptyIdents
+        identity = emptyIdents[idIndex]
+        if idIndex < emptyIdents.length - 1
+          PGPKeyStore.getKeyContents(key: identity, callback: (identity) => newIdents.push(identity))
+        else
+          PGPKeyStore.getKeyContents(key: identity, callback: (identity) =>
+            newIdents.push(identity)
+            @props.callback(newIdents)
```

Comment:
Hmm, I'm not sure about the intended behavior here. It looks like this will call props.callback three times if there are three empty idents, once with a [A], again with [A, B], and then again with [A,B,C]. Maybe we should use an `async` library method to collect all of the results from `getKeyContents` before firing the callback with all of them?